store/mockstore/mocktikv/mvcc.go

```diff
@@ -217,17 +217,23 @@ func (l *mvccLock) check(ts uint64, key []byte) (uint64, error) {
 	if ts == math.MaxUint64 && bytes.Equal(l.primary, key) {
 		return l.startTS - 1, nil
 	}
+	// Skip lock if the lock is resolved.
+	for _, resolved := range resolvedLocks {
+		if l.startTS == resolved {
+			return l.startTS, nil
```

Comment:
We should return the original TS because the lock.startTS may be smaller than the latest commit TS and smaller than read ts.

We return the ts only for `MaxUint64`.